recipes/dd-handler.rb

```diff
@@ -46,21 +43,36 @@
   ENV['DATADOG_PROXY'] = proxy_url.to_s
 end
 
-handler_config = {
-  :api_key => node['datadog']['api_key'],
-  :application_key => node['datadog']['application_key'],
-  :use_ec2_instance_id => node['datadog']['use_ec2_instance_id'],
-  :tag_prefix => node['datadog']['tag_prefix']
-}
-
-unless node['datadog']['use_ec2_instance_id']
-  handler_config[:hostname] = node['datadog']['hostname']
-end
-
 # Create the handler to run at the end of the Chef execution
 chef_handler 'Chef::Handler::Datadog' do
+  def handler_config # rubocop:disable Metrics/AbcSize
+    config = {
+      :api_key => Chef::Datadog.api_key(node),
+      :application_key => Chef::Datadog.application_key(node),
+      :use_ec2_instance_id => node['datadog']['use_ec2_instance_id'],
+      :tag_prefix => node['datadog']['tag_prefix']
+    }
+    unless node['datadog']['use_ec2_instance_id']
+      config[:hostname] = node['datadog']['hostname']
+    end
+    config
+  end
+
+  require 'chef/handler/datadog'
```

Comment:
@iancward Note the move of the `require` line to here.  Since it's now declared in a resource provider block, it isn't evaluated until runtime.  Therefore, the Gem no longer needs to be installed at compile time.  As long as it's declared before this handler resource is declared, things will work properly even if the Gem is installed at runtime.